### ISSUE TITLE
Server crashed after I installed the latest version 

### ISSUE THREAD

**Issue by mkawczynski07** [Alex-MacLean/TheLegendOfHerobrine#4]
Server:
```
-- System Details --
Details:
	Minecraft Version: 1.15.2
	Minecraft Version ID: 1.15.2
	Operating System: Linux (amd64) version 4.15.0-29-generic
	Java Version: 10.0.2, Oracle Corporation
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 3950824352 bytes (3767 MB) / 4294967296 bytes (4096 MB) up to 4294967296 bytes (4096 MB)
	CPUs: 12
	JVM Flags: 2 total; -Xmx4096M -Xms4096M
	ModLauncher: 5.1.0+69+master.79f13f7
	ModLauncher launch target: fmlserver
	ModLauncher naming: srg
	ModLauncher services: 
		/eventbus-2.1.0-service.jar eventbus PLUGINSERVICE 
		/forge-1.15.2-31.1.63.jar object_holder_definalize PLUGINSERVICE 
		/forge-1.15.2-31.1.63.jar runtime_enum_extender PLUGINSERVICE 
		/forge-1.15.2-31.1.63.jar capability_inject_definalize PLUGINSERVICE 
		/accesstransformers-2.1.1-shadowed.jar accesstransformer PLUGINSERVICE 
		/forge-1.15.2-31.1.63.jar runtimedistcleaner PLUGINSERVICE 
		/forge-1.15.2-31.1.63.jar fml TRANSFORMATIONSERVICE 
	FML: 31.1
	Forge: net.minecraftforge:31.1.63
	FML Language Providers: 
		javafml@31.1
		minecraft@1
	Mod List: 
		forge-1.15.2-31.1.63-server.jar Minecraft {minecraft@1.15.2 LOAD_REGISTRIES}
		Useful-Backpacks-Mod-1.15.2.jar Useful Backpacks {usefulbackpacks@1.9.0.52 LOAD_REGISTRIES}
		OfflineSkins-1.15.2-v1.jar OfflineSkins {offlineskins@1.15.2-v1 LOAD_REGISTRIES}
		forgemod_VoxelMap-1.9.16_for_1.15.2.jar VoxelMap {voxelmap@1.9.16 LOAD_REGISTRIES}
		forge-1.15.2-31.1.63-universal.jar Forge {forge@31.1.63 LOAD_REGISTRIES}
		BetterStorageToo-1.15.2-5.0.0.1.jar BetterStorageToo {betterstorage@1.15.2-5.0.0.1 LOAD_REGISTRIES}
		U-Team-Core-1.15.2.jar U Team Core {uteamcore@2.10.0.146 LOAD_REGISTRIES}
		LegendOfHerobrine-1.15.2-0.4.6.jar The Legend of Herobrine {herobrine@0.4.6 ERROR}
		Inventory-Tweaks-Reborn-Mod-1.15.2.jar Inventory Tweaks Reborn {invtweaks@1.15.2-1.0.1 LOAD_REGISTRIES}
		damage_indicator-0.0.9.jar Ama's Damage Indicator {damage_indicator@0.0.9 LOAD_REGISTRIES}
		AdvancedSkinCustomization-2.8.2.jar Advanced Skin Customization {advskinmod@2.8.2 LOAD_REGISTRIES}
	Data Packs: 
	Is Modded: Definitely; Server brand changed to 'forge'
	Type: Dedicated Server (map_server.txt)
```

Exception, just after start the server:

```
[08:15:16] [Server thread/FATAL] [cp.mo.mo.TransformerClassWriter/]: Class net/minecraft/entity/IRendersAsItem unable to find resource null
[08:15:16] [Server thread/ERROR] [ne.mi.fm.ja.FMLModContainer/]: Exception caught during firing event: Failed to load hierarchy member net/minecraft/entity/IRendersAsItem
        Index: 4
        Listeners:
                0: NORMAL
                1: net.minecraftforge.eventbus.EventBus$$Lambda$2927/1432780713@6a0fe8de
                2: ASM: class com.herobrine.mod.HerobrineMod$RegistryEvents registerBiomes(Lnet/minecraftforge/event/RegistryEvent$Register;)V
                3: ASM: class com.herobrine.mod.HerobrineMod$RegistryEvents registerBlocks(Lnet/minecraftforge/event/RegistryEvent$Register;)V
                4: ASM: class com.herobrine.mod.HerobrineMod$RegistryEvents registerItems(Lnet/minecraftforge/event/RegistryEvent$Register;)V
                5: ASM: class com.herobrine.mod.HerobrineMod$RegistryEvents registerEntities(Lnet/minecraftforge/event/RegistryEvent$Register;)V
java.lang.RuntimeException: Failed to load hierarchy member net/minecraft/entity/IRendersAsItem
        at cpw.mods.modlauncher.TransformerClassWriter.computeHierarchy(TransformerClassWriter.java:86)
        at cpw.mods.modlauncher.TransformerClassWriter.access$100(TransformerClassWriter.java:35)
        at cpw.mods.modlauncher.TransformerClassWriter$SuperCollectingVisitor.lambda$visit$0(TransformerClassWriter.java:136)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
        at cpw.mods.modlauncher.TransformerClassWriter$SuperCollectingVisitor.visit(TransformerClassWriter.java:135)
        at org.objectweb.asm.tree.ClassNode.accept(ClassNode.java:340)
        at cpw.mods.modlauncher.TransformerClassWriter.computeHierarchy(TransformerClassWriter.java:74)
        at cpw.mods.modlauncher.TransformerClassWriter.<init>(TransformerClassWriter.java:48)
        at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120)
        at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:239)
        at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:126)
        at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:96)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:499)
        at com.herobrine.mod.HerobrineMod$RegistryEvents.registerItems(HerobrineMod.java:89)
        at net.minecraftforge.eventbus.ASMEventHandler_19_RegistryEvents_registerItems_Register.invoke(.dynamic)
        at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:80)
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258)
        at net.minecraftforge.fml.javafmlmod.FMLModContainer.fireEvent(FMLModContainer.java:106)
        at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
        at java.base/java.util.function.Consumer.lambda$andThen$0(Consumer.java:65)
        at net.minecraftforge.fml.ModContainer.transitionState(ModContainer.java:112)
        at net.minecraftforge.fml.ModList.lambda$dispatchSynchronousEvent$5(ModList.java:126)
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1492)
        at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658)
        at net.minecraftforge.fml.ModList.dispatchSynchronousEvent(ModList.java:126)
        at net.minecraftforge.fml.ModList.lambda$static$1(ModList.java:96)
        at net.minecraftforge.fml.LifecycleEventProvider.dispatch(LifecycleEventProvider.java:71)
        at net.minecraftforge.fml.ModLoader.dispatchAndHandleError(ModLoader.java:214)
        at net.minecraftforge.fml.ModLoader.lambda$gatherAndInitializeMods$25(ModLoader.java:206)
        at net.minecraftforge.registries.GameData.fireRegistryEvents(GameData.java:971)
        at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:206)
        at net.minecraftforge.fml.server.ServerModLoader.begin(ServerModLoader.java:45)
        at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:122)
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:613)
        at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.io.IOException: Class not found
        at org.objectweb.asm.ClassReader.readStream(ClassReader.java:310)
        at org.objectweb.asm.ClassReader.<init>(ClassReader.java:283)
        at cpw.mods.modlauncher.TransformerClassWriter.computeHierarchy(TransformerClassWriter.java:82)
        ... 35 more

[08:15:16] [Server thread/ERROR] [ne.mi.fm.ja.FMLModContainer/LOADING]: Caught exception during event RegistryEvent.Register<minecraft:item> dispatch for modid herobrine
java.lang.RuntimeException: Failed to load hierarchy member net/minecraft/entity/IRendersAsItem
        at cpw.mods.modlauncher.TransformerClassWriter.computeHierarchy(TransformerClassWriter.java:86) ~[modlauncher-5.1.0.jar:?] {}
        at cpw.mods.modlauncher.TransformerClassWriter.access$100(TransformerClassWriter.java:35) ~[modlauncher-5.1.0.jar:?] {}
        at cpw.mods.modlauncher.TransformerClassWriter$SuperCollectingVisitor.lambda$visit$0(TransformerClassWriter.java:136) ~[modlauncher-5.1.0.jar:?] {}
        at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948) ~[?:?] {}
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658) ~[?:?] {}
        at cpw.mods.modlauncher.TransformerClassWriter$SuperCollectingVisitor.visit(TransformerClassWriter.java:135) ~[modlauncher-5.1.0.jar:?] {}
        at org.objectweb.asm.tree.ClassNode.accept(ClassNode.java:340) ~[asm-tree-7.2.jar:7.2] {}
        at cpw.mods.modlauncher.TransformerClassWriter.computeHierarchy(TransformerClassWriter.java:74) ~[modlauncher-5.1.0.jar:?] {}
        at cpw.mods.modlauncher.TransformerClassWriter.<init>(TransformerClassWriter.java:48) ~[modlauncher-5.1.0.jar:?] {}
        at cpw.mods.modlauncher.ClassTransformer.transform(ClassTransformer.java:120) ~[modlauncher-5.1.0.jar:?] {}
        at cpw.mods.modlauncher.TransformingClassLoader$DelegatedClassLoader.findClass(TransformingClassLoader.java:239) ~[modlauncher-5.1.0.jar:?] {}
        at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:126) ~[modlauncher-5.1.0.jar:?] {}
        at cpw.mods.modlauncher.TransformingClassLoader.loadClass(TransformingClassLoader.java:96) ~[modlauncher-5.1.0.jar:?] {}
        at java.lang.ClassLoader.loadClass(ClassLoader.java:499) ~[?:?] {}
        at com.herobrine.mod.HerobrineMod$RegistryEvents.registerItems(HerobrineMod.java:89) ~[?:1.15.2-0.4.6] {re:classloading}
        at net.minecraftforge.eventbus.ASMEventHandler_19_RegistryEvents_registerItems_Register.invoke(.dynamic) ~[?:?] {}
        at net.minecraftforge.eventbus.ASMEventHandler.invoke(ASMEventHandler.java:80) ~[eventbus-2.1.0-service.jar:?] {}
        at net.minecraftforge.eventbus.EventBus.post(EventBus.java:258) ~[eventbus-2.1.0-service.jar:?] {}
        at net.minecraftforge.fml.javafmlmod.FMLModContainer.fireEvent(FMLModContainer.java:106) ~[?:31.1] {re:classloading}
        at java.util.function.Consumer.lambda$andThen$0(Consumer.java:65) ~[?:?] {}
        at java.util.function.Consumer.lambda$andThen$0(Consumer.java:65) ~[?:?] {}
        at net.minecraftforge.fml.ModContainer.transitionState(ModContainer.java:112) ~[?:?] {re:classloading}
        at net.minecraftforge.fml.ModList.lambda$dispatchSynchronousEvent$5(ModList.java:126) ~[?:?] {re:classloading}
        at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1492) ~[?:?] {}
        at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:658) ~[?:?] {}
        at net.minecraftforge.fml.ModList.dispatchSynchronousEvent(ModList.java:126) ~[?:?] {re:classloading}
        at net.minecraftforge.fml.ModList.lambda$static$1(ModList.java:96) ~[?:?] {re:classloading}
        at net.minecraftforge.fml.LifecycleEventProvider.dispatch(LifecycleEventProvider.java:71) ~[?:?] {re:classloading}
        at net.minecraftforge.fml.ModLoader.dispatchAndHandleError(ModLoader.java:214) ~[?:?] {re:classloading}
        at net.minecraftforge.fml.ModLoader.lambda$gatherAndInitializeMods$25(ModLoader.java:206) ~[?:?] {re:classloading}
        at net.minecraftforge.registries.GameData.fireRegistryEvents(GameData.java:971) [?:?] {re:classloading}
        at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:206) [?:?] {re:classloading}
        at net.minecraftforge.fml.server.ServerModLoader.begin(ServerModLoader.java:45) [?:?] {re:classloading}
        at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:122) [?:?] {re:classloading,pl:accesstransformer:B}
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:613) [?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
        at java.lang.Thread.run(Thread.java:844) [?:?] {}
Caused by: java.io.IOException: Class not found
        at org.objectweb.asm.ClassReader.readStream(ClassReader.java:310) ~[asm-7.2.jar:7.2] {}
        at org.objectweb.asm.ClassReader.<init>(ClassReader.java:283) ~[asm-7.2.jar:7.2] {}
        at cpw.mods.modlauncher.TransformerClassWriter.computeHierarchy(TransformerClassWriter.java:82) ~[modlauncher-5.1.0.jar:?] {}
        ... 35 more
[08:15:16] [Server thread/FATAL] [ne.mi.fm.ModLoader/LOADING]: Failed to complete lifecycle event LOAD_REGISTRIES, 1 errors found
[08:15:16] [Server thread/ERROR] [minecraft/MinecraftServer]: Encountered an unexpected exception
net.minecraftforge.fml.LoadingFailedException: Loading errors encountered: [
        The Legend of Herobrine (herobrine) encountered an error during the load_registries event phase
§7Failed to load hierarchy member net/minecraft/entity/IRendersAsItem
]
        at net.minecraftforge.fml.ModLoader.dispatchAndHandleError(ModLoader.java:218) ~[?:?] {re:classloading}
        at net.minecraftforge.fml.ModLoader.lambda$gatherAndInitializeMods$25(ModLoader.java:206) ~[?:?] {re:classloading}
        at net.minecraftforge.registries.GameData.fireRegistryEvents(GameData.java:971) ~[?:?] {re:classloading}
        at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:206) ~[?:?] {re:classloading}
        at net.minecraftforge.fml.server.ServerModLoader.begin(ServerModLoader.java:45) ~[?:?] {re:classloading}
        at net.minecraft.server.dedicated.DedicatedServer.func_71197_b(DedicatedServer.java:122) ~[?:?] {re:classloading,pl:accesstransformer:B}
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:613) [?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
        at java.lang.Thread.run(Thread.java:844) [?:?] {}
[08:15:16] [Server thread/ERROR] [minecraft/MinecraftServer]: This crash report has been saved to: /opt/minecraft_copy/./crash-reports/crash-2020-05-22_08.15.16-server.txt
[08:15:16] [Server thread/INFO] [minecraft/MinecraftServer]: Stopping server
[08:15:16] [Server thread/INFO] [minecraft/MinecraftServer]: Saving worlds
[08:15:16] [Server thread/ERROR] [minecraft/MinecraftServer]: Exception stopping the server
java.lang.RuntimeException: The server has failed to initialize correctly due to mod loading errors. Examine the crash report for more details.
        at net.minecraftforge.common.DimensionManager.getWorld(DimensionManager.java:185) ~[?:?] {re:classloading}
        at net.minecraft.server.MinecraftServer.func_71218_a(MinecraftServer.java:1004) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
        at net.minecraft.server.MinecraftServer.func_213211_a(MinecraftServer.java:536) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
        at net.minecraft.server.MinecraftServer.func_71260_j(MinecraftServer.java:568) ~[?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
        at net.minecraft.server.dedicated.DedicatedServer.func_71260_j(DedicatedServer.java:550) ~[?:?] {re:classloading,pl:accesstransformer:B}
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:676) [?:?] {re:classloading,pl:accesstransformer:B,pl:runtimedistcleaner:A}
        at java.lang.Thread.run(Thread.java:844) [?:?] {}
```

**Comment by Alex-MacLean** [comment]
It is a known issue that the latest 1.15.2 builds crash the server. I am looking into the cause and I will be releasing a patch with a fix for this very soon.

Also try updating the forge server software and using JRE 8. This mod does cause server crashes, but the older versions of Forge are buggy, and Forge has problems with Java versions above 8.

Update: I found the problem! The entities for Holy Water and Unholy water implement IRenderAsItem. Because of how Forge works it loads that on the server and causes a crash. Removing the implement fixes the issue and doesn't seem to cause any apparent problems.

**Comment by mkawczynski07** [comment]
@Alex-MacLean  good to know. I don't know Minecraft and forge API but I saw that the problem is with registering those items. When do you plan release fix?

**Comment by Alex-MacLean** [comment]
This bug is only in the 1.15.2 build, it was leftover from when I was optimizing and improving the code in 0.4.4. I am currently investigating a rare bug that was supposed to be fixed in the 0.4.4 update that causes Herobrine to not spawn after an altar is activated. I plan on releasing a build 2 of 0.4.6 later today after I confirm I didn't break anything since this is such a small change, but 0.4.7 may still happen if the spawning glitches still exists.

**Comment by mkawczynski07** [comment]
Works! Thx!
